plugins/out_stackdriver/stackdriver.c

```diff
@@ -530,6 +530,38 @@ static int process_local_resource_id(const void *data, size_t bytes,
     return -1;
 }
 
+/*
+ * parse_labels
+ * - Iterate throught the original payload (obj) and find out the entry that matches
+ *   the labels_key 
+ * - Used to convert all labels under labels_key to root-level `labels` field
+ */
+static msgpack_object *parse_labels(struct flb_stackdriver *ctx, msgpack_object *obj)
+{
+    int i;
+    int len;
+    msgpack_object_kv *kv = NULL;
+
+    if (!obj || obj->type != MSGPACK_OBJECT_MAP) {
+        return NULL;
+    }
+
+    len = flb_sds_len(ctx->labels_key);
+    for (i = 0; i < obj->via.map.size; i++) {
+        kv = &obj->via.map.ptr[i];
+        if (flb_sds_casecmp(ctx->labels_key, kv->key.via.str.ptr, len) == 0) {
+            if (kv->val.type != MSGPACK_OBJECT_MAP) {
+                flb_plg_warn(ctx->ins, "the type of labels should be map");
```

Comment:
What would happen if the labels are not in a map?

Reply:
If labels are not in a map but is a str. Then this will be rejected with error:
```
{
  "error": {
    "code": 400,
    "message": "Invalid value at 'entries[0].labels' (type.googleapis.com/google.logging.v2.LogEntry.LabelsEntry), \"testLabels\"",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "field": "entries[0].labels",
            "description": "Invalid value at 'entries[0].labels' (type.googleapis.com/google.logging.v2.LogEntry.LabelsEntry), \"testLabels\""
          }
        ]
      }
    ]
  }
}
```

Reply:
Note: Parameter validation should terminate the process. It should be flb_plg_error() and throw proper error.